src/pytest_plugins/filler/static_filler.py

```diff
@@ -206,7 +207,39 @@ def collect(self: "FillerFile") -> Generator["FillerTestItem", None, None]:
                         and (mark.name not in [v.mark.name for v in validity_markers])
                     ]
 
-                    for format_with_or_without_label in fixture_formats:
+                    # Apply fixture format filtering for pre-allocation groups
+                    generate_pre_alloc = self.config.getoption("generate_pre_alloc_groups")
+                    use_pre_alloc = self.config.getoption("use_pre_alloc_groups")
+                    if generate_pre_alloc or use_pre_alloc:
+                        # When pre-allocation group flags are set, only generate
+                        # BlockchainEngineXFixture
+                        filtered_formats = [
+                            format_item
+                            for format_item in fixture_formats
+                            if (
+                                format_item is BlockchainEngineXFixture
+                                or (
+                                    isinstance(format_item, LabeledFixtureFormat)
+                                    and format_item.format is BlockchainEngineXFixture
+                                )
+                            )
+                        ]
+                    else:
+                        # Filter out BlockchainEngineXFixture if pre-allocation group
+                        # flags not set
+                        filtered_formats = [
+                            format_item
+                            for format_item in fixture_formats
+                            if not (
+                                format_item is BlockchainEngineXFixture
+                                or (
+                                    isinstance(format_item, LabeledFixtureFormat)
+                                    and format_item.format is BlockchainEngineXFixture
+                                )
+                            )
+                        ]
+
+                    for format_with_or_without_label in filtered_formats:
```

Comment:
I think this was copied verbatim from `filler.py`. Perhaps we can add a helper function, but perhaps it's not worth it?

Reply:
I can look into DRYing this up before we're ready to merge 👍🏼